chain/network/src/routing/edge_set.rs

```diff
@@ -0,0 +1,59 @@
+use crate::routing::Edge;
+use near_primitives::borsh::maybestd::borrow::Borrow;
+use near_primitives::network::PeerId;
+use std::collections::hash_set::Iter;
+use std::collections::HashSet;
+use std::hash::{Hash, Hasher};
+
+/// Wraps around `Edge` struct. The main feature of this struct, is that it's hashed by
+/// `(Edge::key.0, Edge::key.1)` pair instead of `(Edge::key.0, Edge::key.1, Edge::nonce)`
+/// triple.
+#[derive(Eq, PartialEq)]
+pub struct EdgeIndexedByKey {
+    pub inner: Edge,
+}
```

Comment:
```suggestion
struct EdgeIndexedByKey {
    inner: Edge,
}
```

Making this private is the main thing I'd love to achieve by moving stuff to a separate module

Reply:
@matklad Yes, I can do that thanks to usage of `impl Iterator`.